### ISSUE TITLE
Tax - Tax name saved disappears after refreshing the page and directed to previous page.

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#45888]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.10
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Pre-condition: Have a workspace with tax enabled
1. Go to https://staging.new.expensify.com/home
2. Tap profile icon -- workspaces -- workspace
3. Tap tax -- Add tax rate -- Name
4. Enter a name and save it
5. Tap name
6. Refresh the page

## Expected Result:
Tax name saved must not disappear after refreshing the page and user must stay in same page and must not directed to previous page.

## Actual Result:
Tax name saved disappears after refreshing the page and directed to previous page.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/32d75bb7-b0c6-4be9-ba6e-3a1577759ed5

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
We think this issue might be related to the  #vip-vsb

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Tax - Tax name saved disappears after refreshing the page and directed to previous page.

### What is the root cause of that problem?
1. tax name saved disappears: because we haven't passed `shouldSaveDraft` on the input wrappers
2. directed to previous page: it is because `TextPicker` only opens modal for inputting text; it doesn't open a new page when clicking on the menu. And the status of the modal 
https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/components/TextPicker/index.tsx#L12-L13
is just a react state and gets refreshed when refreshing page
### What changes do you think we should make in order to solve the problem?

1. we should pass `shouldSaveDraft={true}` for both input wrappers in the [WorkspaceCreateTaxPage](https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/pages/workspace/taxes/WorkspaceCreateTaxPage.tsx#L90-L113)
And additionally we might need to clear the draft values 
```
            clearDraftValues(ONYXKEYS.FORMS.WORKSPACE_NEW_TAX_FORM)

```
[`onSubmit`](https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/pages/workspace/taxes/WorkspaceCreateTaxPage.tsx#L40-L47) of the form or even on the closing of the page (using `useEffect` on the unmounting cycle)
2. The second problem is not specific to tax but with `TextPicker` I don't think we should alter the behavior but if we need we can do by saving the modal open state in onyx 
### What alternative solutions did you explore? (Optional)

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Tax - Tax name saved disappears after refreshing the page and directed to previous page.

### What is the root cause of that problem?
1. tax name saved disappears: because we haven't passed `shouldSaveDraft` on the input [wrappers](https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/pages/workspace/taxes/WorkspaceCreateTaxPage.tsx#L90-L113)
2. directed to previous page: it is because `TextPicker` only opens modal for inputting text instead of openning a new page when clicking on the menu. And the open status of the modal 
https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/components/TextPicker/index.tsx#L12-L13
is just a react state and gets refreshed when refreshing page
### What changes do you think we should make in order to solve the problem?

1. we should pass `shouldSaveDraft={true}` for both input wrappers in the [WorkspaceCreateTaxPage](https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/pages/workspace/taxes/WorkspaceCreateTaxPage.tsx#L90-L113)
And additionally, we might need to clear the draft values 
```
            clearDraftValues(ONYXKEYS.FORMS.WORKSPACE_NEW_TAX_FORM)

```
[`onSubmit`](https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/pages/workspace/taxes/WorkspaceCreateTaxPage.tsx#L40-L47) of the form or even on the pressing of the page's [back button](https://github.com/Expensify/App/blob/b8b99d0a994e3ccf8b91564df0f92a7c440aa6c7/src/pages/workspace/taxes/WorkspaceCreateTaxPage.tsx#L80-L81)
2. The second problem is not specific to tax but with `TextPicker` I don't think we should alter the behavior but if we need we can do by saving the modal open state in onyx by giving id's to text pickers and also clearing the state when necessary. But it an expected behaviour IMO.
### What alternative solutions did you explore? (Optional)

**Comment by trjExpensify** [comment]
I wouldn't really say this is "tax named saved" they're refreshing the page before saving the input. I think this is expected on these edit pages with a TextPicker. 

I'm inclined to close this issue. @mountiny @luacmartins let me know if you think any differently.

**Comment by mountiny** [comment]
yeah I think this might be occurring across the various flows, I agree we can close but curious to here from @luacmartins as he worked with the forms more closely

**Comment by luacmartins** [comment]
We can save the draft if we want to, it would be pretty easy since this page uses the form component. Got a draft PR to do that.

**Comment by luacmartins** [comment]
I can take a look at the other flows tomorrow and make sure they all save the drafts

**Comment by trjExpensify** [comment]
Yeah, I just tested all of members, distance rates, categories, tags and report fields. We don't save any drafts. If that's the consistent thing to be doing with the form component, then let's do it. 

That said, I just looked at the `Add debit card` form and `Status` input, and we don't save the draft input there either, it's lost if you don't click `Save` and refresh. 

So I'm inclined to say let's do nothing, tbh.
